### PR TITLE
WIP: Added PodGroup API for coscheduling

### PR DESCRIPTION
**Pull request by k82cn**
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Part of #61012

**Release note**:
```release-note
Added PodGroup API for Gang-Scheduling
```

### CODE REVIEW

**Comment by k82cn** [comment]
/hold

Please hold the review until https://github.com/kubernetes/community/pull/2337 merged.

**Comment by tpepper** [comment]
At this late point for a size/XXL and implementation in WIP state, I'm worried about this for v1.12 inclusion.  Neither this PR or https://github.com/kubernetes/kubernetes/issues/61012 are marked for the milestone, but https://github.com/kubernetes/features/issues/583 is.  Should this be removed from the milestone?

**Comment by k82cn** [comment]
@tpepper , considering the progress of design doc review, prefer to move this to 1.13; also append this suggestion to the feature. Thanks for your attention.

File: pkg/apis/scheduling/types.go [hunk 1]
```diff
@@ -79,3 +82,42 @@ type PriorityClassList struct {
...
+
+// PodGroupSpec is a description of a pod group.
+type PodGroupSpec struct {
+	// NumMember defines the number of members/tasks to run the pod group;
+	// if there's not enough resources to start all tasks, the scheduler
+	// will not start anyone.
+	NumMember int32
+	// Resources defines the total resource the PodGroup requests to run
+	// Pods.
+	Resources map[string]resource.Quantity
```
Comment (k82cn) [inline pkg/apis/scheduling/types.go]: @liggitt  ,  do you know how to use `v1.ResourceList` here? The `make update` did not include `k8s.io/apimachinery/pkg/api/resource` in `generated.pb.go`.

```bash
vendor/k8s.io/api/scheduling/v1alpha1/generated.pb.go:850:18: undefined: k8s_io_apimachinery_pkg_api_resource
vendor/k8s.io/api/scheduling/v1alpha1/generated.pb.go:857:18: undefined: k8s_io_apimachinery_pkg_api_resource
```

**Pull request closed without merging**